lib/teiserver/battle/libs/balance_lib.ex

```diff
@@ -94,11 +106,31 @@ defmodule Teiserver.Battle.BalanceLib do
       groups
       |> Enum.map(fn members ->
         userids = Map.keys(members)
-        ratings = Map.values(members)
+        ratings = Map.values(members) |> Enum.map(fn x -> x.rating end)
+
+        ranks =
+          Map.values(members)
+          |> Enum.map(fn x ->
+            cond do
+              Map.has_key?(x, :rank) -> x.rank
+              true -> 0
```

Comment:
You can use a shorter version: `Map.get(x, :rank, 0)`

Reply:
Nice tip. Will use this one in the future.